File: back/src/controller/logout.controller.js
const db = require('../db/conection');

const postLogout = async (req, res) => {
  try {
    if (req.session && req.session.id_user) {
      req.session.destroy((err) => {
        if (err) {
          console.error('Error al destruir la sesión:', err);
          return res.status(500).json({ error: 'Error interno del servidor' });
        }
        res.status(200).json({ mensaje: 'Sesión cerrada correctamente' });
      });
    } else {
      res.status(400).json({ error: 'No hay sesión activa' });
    }
  } catch (error) {
    console.error('Error en el logout:', error);
    res.status(500).json({ error: 'Error interno del servidor' });
  }
};

module.exports = postLogout;

File: back_py/controller/logout_controller.py
# Controlador de logout (versión Python, Flask)
# En Python se recomienda snake_case para nombres de archivos y funciones
# Este controlador implementa la lógica para cerrar la sesión del usuario

from flask import session  # Importa el objeto session de Flask

def post_logout():
    try:
        # Verifica si hay sesión activa (en JS: req.session && req.session.usuarioId)
        if session.get('usuario_id'):
            # Elimina los datos de la sesión (en JS: req.session.destroy())
            session.clear()  # Borra toda la sesión en Flask
            return {'mensaje': 'Sesión cerrada correctamente'}, 200
        else:
            return {'error': 'No hay sesión activa'}, 400
    except Exception as e:
        print('Error en el logout:', e)
        return {'error': 'Error interno del servidor'}, 500
